core/vm/evm.go

```diff
@@ -48,6 +48,8 @@ func run(evm *EVM, contract *Contract, input []byte, readOnly bool) ([]byte, err
 	if contract.CodeAddr != nil {
 		var precompiles map[common.Address]PrecompiledContract
 		switch {
+		case evm.chainRules.IsShanghai:
```

Comment:
It's not a good idea to use `evm.chainRules.IsShanghai` here although `TIPXDCXDISABLE` equals to `ShanghaiBlock `.

Reply:
yes, would you suggest 
`case evm.chainConfig.IsTIPXDCXMiner(evm.BlockNumber):`
like the below condition?

Reply:
like `evm.chainRules.IsXdcxDisable`

Reply:
fixed